app/views/search/index.html.erb

```diff
@@ -0,0 +1,52 @@
+<% title search_query %>
+<% noindex %>
+
+<div class="max-w-screen-xl mx-auto md:px-6 px-3 pt-20">
+  <div class="md:p-3 py-3 pb-5 border-b border-gray-200">
+    <h1 class="md:text-3xl text-xl font-semibold text-gray-800 dark:text-gray-200">
+      Results for: "<%= search_query %>"
+    </h1>
+    <% if Rails.env.development? %>
+      <details class="border border-gray-200 p-3 mt-1 bg-gray-100 dark:bg-gray-700 dark:border-gray-500">
+        <summary class="focus:outline-none">ElasticSearch Query</summary>
+        <pre><%= JSON.pretty_generate @search.query %></pre>
+      </details>
+    <% end %>
+  </div>
+  <% @search.results.each do |r| %>
+    <div class="md:p-3 py-3 my-1 rounded dark:text-gray-500">
+      <div class="flex">
+        <div class="w-full">
+          <% if r.is_a? RubyMethod %>
```

Comment:
Since both halves of this are identical other than the method, it feels like this could get DRYed up with a little duck typing.

If a new method was added to all of possible types (name or something generic), then you could just call that method and it'd print the appropriate thing for the type.

Reply:
I agree with all your points, and honestly this whole page has been neglected and badly needs a bit of a refresh.